lib/kamal/commands/app.rb

```diff
@@ -234,6 +234,6 @@ def find_and_remove_older_siblings(path)
     end
 
     def copy_contents(source, destination, continue_on_error: false)
-      [ :cp, "-rn", "#{source}/*", destination, *("|| true" if continue_on_error)]
+      [ :cp, "-rnT", "#{source}", destination, *("|| true" if continue_on_error)]
```

Comment:
Looks like `-T` is GNU-only. Copying "#{source}/.", "#{destination}/." could be a workaround, avoiding the differing trailing-slash behavior in BSD vs GNU.